### PR TITLE
Allow `GET /api/dataset_collections/{hdca_id}` without specifying `instance_type`

### PR DESCRIPTION
**Pull request by nsoranzo**
Also:
- raise `exceptions.NotImplemented` for `GET /api/dataset_collections`
- add/fix API docs

### CODE REVIEW

File: lib/galaxy/webapps/galaxy/api/dataset_collections.py [hunk 1]
```diff
@@ -63,14 +62,18 @@ def create(self, trans: ProvidesHistoryContext, payload: dict, **kwd):
...
         dataset_collection_instance = self.__service.create(trans=trans, **create_params)
         return dictify_dataset_collection_instance(dataset_collection_instance,
                                                    security=trans.security, parent=create_params["parent"])
 
     @expose_api
-    def show(self, trans: ProvidesHistoryContext, instance_type, id, **kwds):
+    def show(self, trans: ProvidesHistoryContext, id, instance_type='history', **kwds):
+        """
+        GET /api/dataset_collections/{hdca_id}
+        GET /api/dataset_collections/{ldca_id}?instance_type=library
```
Comment (mvdbeek) [inline lib/galaxy/webapps/galaxy/api/dataset_collections.py:75]: ldca's are not really a galaxy type, these are still hdcas, but the parent is a library.
```suggestion
        GET /api/dataset_collections/{hdca_id}?instance_type=library
```
Seems more accurate

Comment (nsoranzo) [reply lib/galaxy/webapps/galaxy/api/dataset_collections.py:75]: We do have `self.model.LibraryDatasetCollectionAssociation` in `lib/galaxy/model/__init__.py` and it seems to be used when creating a collection with `POST /api/dataset_collections` and `instance_type=library` in the payload (see also `lib/galaxy/managers/collections.py`).

Comment (mvdbeek) [reply lib/galaxy/webapps/galaxy/api/dataset_collections.py:75]: Thanks, I did not know that ...

**Review by mvdbeek — APPROVED** [review]

File: lib/galaxy/webapps/galaxy/api/dataset_collections.py (hunk 1 above)
Comment (nsoranzo) [reply lib/galaxy/webapps/galaxy/api/dataset_collections.py:75]: NP, to be fair [it is not yet possible to create or add collections to libraries in the UI](https://github.com/galaxyproject/galaxy/issues/11250), so...

**Pull request merged**